packages/core/content-manager/admin/src/components/ConfigurationForm/Fields.tsx

```diff
@@ -176,131 +221,246 @@ const Fields = ({ attributes, fieldSizes, components, metadatas = {} }: FieldsPr
-                    name={`layout.${rowIndex}.children.${fieldIndex}`}
-                    onMoveField={handleMoveField}
-                    onRemoveField={handleRemoveField(rowIndex, fieldIndex)}
-                  />
-                </Grid.Item>
-              ))}
-            </Grid.Root>
-          ))}
-          <Menu.Root>
-            <Menu.Trigger
-              startIcon={<Plus />}
-              endIcon={null}
-              disabled={remainingFields.length === 0}
-              fullWidth
-              variant="secondary"
-            >
-              {formatMessage({
-                id: getTranslation('containers.SettingPage.add.field'),
-                defaultMessage: 'Insert another field',
-              })}
-            </Menu.Trigger>
-            <Menu.Content>
-              {remainingFields.map((field) => (
-                <Menu.Item key={field.name} onSelect={handleAddField(field)}>
-                  {field.label}
-                </Menu.Item>
-              ))}
-            </Menu.Content>
-          </Menu.Root>
-        </Flex>
-      </Box>
-    </Flex>
-  );
-};
+    <DndContext
+      onDragStart={(event) => {
+        const containersAsDictionary = Object.fromEntries(
+          containers.map((container) => [container.dndId, container])
+        );
 
-/**
- * @internal
- * @description Small abstraction to solve within an array of fields where you can
- * add a field to the beginning or start, move back and forth what it's index range
- * should be when calculating it's new temp key
- */
-const generateNKeysBetween = <Field extends { __temp_key__: string }>(
-  field: Field[],
-  count: number,
-  currInd: number,
-  newInd: number
-) => {
-  const startKey = currInd > newInd ? field[newInd - 1]?.__temp_key__ : field[newInd]?.__temp_key__;
-  const endKey = currInd > newInd ? field[newInd]?.__temp_key__ : field[newInd + 1]?.__temp_key__;
-
-  return generateNKeysBetweenImpl(startKey, endKey, count);
-};
+        const activeContainer = findContainer(event.active.id, containersAsDictionary);
 
-/**
- * @internal
- * @description chunks a row of layouts by the max size we allow, 12. It does not add the
- * spacers again, that should be added separately.
- */
-const chunkArray = (array: FormField[]) => {
-  const result: Array<FormField[]> = [];
-  let temp: FormField[] = [];
-
-  array.reduce((acc, field) => {
-    if (acc + field.size > 12) {
-      result.push(temp);
-      temp = [field];
-      return field.size;
-    } else {
-      temp.push(field);
-      return acc + field.size;
-    }
-  }, 0);
+        if (!activeContainer) return;
 
-  if (temp.length > 0) {
-    result.push(temp);
-  }
+        const activeItem = getItemFromContainer(
+          event.active.id,
+          containersAsDictionary[activeContainer]
+        );
+
+        if (activeItem) {
+          setActiveDragItem(activeItem);
+        }
+      }}
+      onDragOver={({ active, over }) => {
```

Comment:
did you see any performance issues with this? there are a lot of loops and it will re-run on every render. No need to change it if it doesn't have a visible impact though